clients/da-vinci-client/src/main/java/com/linkedin/davinci/repository/NativeMetadataRepository.java

```diff
@@ -494,11 +494,9 @@ protected Store putStore(Store newStore) {
     // Workaround to make old metadata compatible with new fields
     newStore.fixMissingFields();
     Store oldStore = subscribedStoreMap.put(newStore.getName(), newStore);
-    if (oldStore == null) {
-      totalStoreReadQuota.addAndGet(newStore.getReadQuotaInCU());
-      notifyStoreCreated(newStore);
```

Comment:
I was thinking about this and I realize that it is a different logic here rather than Venice server. 
Here Da Vinci uses this NativeMetadataRepository which is a SubscriptionBasedReadOnlyStoreRepository. Meaning unless the user (DaVinciBackend) asks it to subscribe store(s), it won't subscribe it.
Here the logic of (oldStore == null) could happen when the first time a store is subscribed and refresh() call loads Store object from corresponding system stores, not when a store is created. 


With new system store design settled internally, maybe in the near future we might get rid of this? Or maybe no because Da Vinci should still not subscribing to all stores but only the specific stores it is interested in.

Reply:
What if the listeners don't handle store creation gracefully on `notifyStoreChanged`, since it is:
```
protected void notifyStoreChanged(Store store) {
    for (StoreDataChangedListener listener: listeners) {
      try {
        listener.handleStoreChanged(store);
      } catch (Throwable e) {
        LOGGER.error("Could not handle store updating event for store: {}", store.getName(), e);
      }
    }
  }
```

Reply:
Discussed offline and realize I was wrong.

My concern was if it should be a part of the interface docs that each listener must handle these cases. But I now think `handleStoreCreated` should only be called on store creation. Not when the component loads that store